jina/optimizers/__init__.py

```diff
@@ -169,4 +175,19 @@ def optimize_flow(self, **kwargs) -> 'ResultProcessor':
             sampler = getattr(optuna.samplers, self._sampler)(seed=self._seed, **kwargs)
         study = optuna.create_study(direction=self._direction, sampler=sampler)
         study.optimize(self._objective, n_trials=self._n_trials)
-        return ResultProcessor(study)
+
+        result_processor = ResultProcessor(study)
+        if self._output_file:
+            result_processor.save_parameters(self._output_file)
+        return result_processor
+
+
+def run_optimizer_cli(args: 'Namespace'):
+    """Used to run the optimizer from command line interface.
+
+    :param args: arguments passed via cli
+    """
+    from .flow_runner import SingleFlowRunner
+    with open(args.uses) as f:
+        optimizer = JAML.load(f)
+    optimizer.optimize_flow()
```

Comment:
I'd rather have the output path in the `args` instead of the .yml and make the result saving here explicit, if the output path is given.

Reply:
You mean defining the directory instead of the file? This would mean the user can not define the file name right?

Reply:
I think is about separating clearly optimization and output results processing.